File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    students = {}
    
    for i in range(n):
        students[i+1] = 1
        
    for student in reserve:
        students[student] += 1     
        
    for student in lost:        
        students[student] -= 1
        print(students)
    
    for student in students:
        if students[student] == 0:        
            if student != 1 and students[student-1] == 2:
                students[student-1] -= 1
                students[student] += 1
            elif student != n and students[student+1] == 2:
                students[student+1] -= 1
                students[student] += 1
    
    for i in students.values():
        if i != 0:
            answer += 1
        
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
  const arr = Array(n).fill(1);

  for (let i of lost) {
    arr[i - 1]--;
  }
  for (let i of reserve) {
    arr[i - 1]++;
  }

  for (let i = 0; i < n; i++) {
    if (arr[i] > 0) continue;
    if (i > 0 && arr[i - 1] > 1) {
      arr[i - 1]--;
      arr[i]++;
    } else if (i < n - 1 && arr[i + 1] > 1) {
      arr[i + 1]--;
      arr[i]++;
    }
  }

  return arr.filter((x) => x !== 0).length;
}
